hazelcast/src/main/java/com/hazelcast/internal/eviction/EvictionPolicyEvaluatorProvider.java

```diff
@@ -63,15 +57,15 @@ public static EvictionPolicyEvaluator getEvictionPolicyEvaluator(EvictionConfigu
             return null;
         }
 
-        EvictionPolicyComparator evictionPolicyComparator = null;
+        EvictionPolicyComparator evictionPolicyComparator;
 
         String evictionPolicyComparatorClassName = evictionConfig.getComparatorClassName();
         if (!StringUtil.isNullOrEmpty(evictionPolicyComparatorClassName)) {
             try {
                 evictionPolicyComparator =
                         ClassLoaderUtil.newInstance(classLoader, evictionPolicyComparatorClassName);
             } catch (Exception e) {
-                ExceptionUtil.rethrow(e);
+                throw ExceptionUtil.rethrow(e);
             }
```

Comment:
Hmm, without it the `EvictionPolicyComparator evictionPolicyComparator = null;` initialization is needed as well, since the compiler cannot see that the catched exception is rethrown. I'll revert those changes, but maybe we can clean this up in another PR, this bugs me for a long time.